cpp/src/join/joining.cu

```diff
@@ -533,6 +534,68 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
```

Comment:
Can this be wrapped into a function w/ documentation? It seems like something that would be useful elsewhere and should be a standalone utility function. Something like "synchronize_categories" or similar.